Fix/UI array outputs rendering

iitzIrFan opened the pull request:
<!-- Thanks for submitting a Pull Request to Kestra. To help us review your contribution, please follow the guidelines below:

- Make sure that your commits follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification e.g. `feat(ui): add a new navigation menu item` or `fix(core): fix a bug in the core model` or `docs: update the README.md`. This will help us automatically generate the changelog.
- The title should briefly summarize the proposed changes.
- Provide a short overview of the change and the value it adds.
- Share a flow example to help the reviewer understand and QA the change.
- Use "closes" to automatically close an issue. For example, `closes #1234` will close issue #1234. -->

### What changes are being made and why?

Closes https://github.com/kestra-io/kestra/issues/10770.

---

### How the changes have been QAed?

Flow used to generate the output.

```yaml
id: array-output-test-simple
namespace: test

tasks:
  - id: generate_array
    type: io.kestra.plugin.core.debug.Return
    format: >
      {
        "simpleArray": [1, 2, 3, 4, 5],
        "arrayOfObjects": [
          { "name": "item1", "value": 100 },
          { "name": "item2", "value": 200 }
        ],
        "toolExecutions": [
          { "tool": "hammer", "duration": 5, "success": true },
          { "tool": "screwdriver", "duration": 10, "success": false }
        ]
      }
   ```

---

### Setup Instructions

<!--If there are any setup requirements like API keys or trial accounts, kindly include brief bullet-points-description outlining the setup process below.

- [External System Documentation](URL)
- Steps to set up the necessary resources

If there are no setup requirements, you can remove this section.

Thank you for your contribution. ❤️  -->

Commit:
fix(outputs): preserve array structure for multiple elements in toolExecutions

Commit:
fix(outputs): enhance array handling in transform function for better structure representation

iitzIrFan committed:
Merge branch 'develop' into fix/ui-array-outputs-rendering

MilosPaunovic committed:
Merge branch 'develop' into fix/ui-array-outputs-rendering

MilosPaunovic reviewed (changes requested):
Hey @iitzIrFan,

I'm not sure I'm able to see what exactly changed here:

Current `develop` branch:

- Selected field:
<img width="1877" height="803" alt="image" src="https://github.com/user-attachments/assets/4cdc0169-4c47-41fa-9e55-fa314eb9f832" />
- Clicked on `Debug expression`:
<img width="1877" height="803" alt="image" src="https://github.com/user-attachments/assets/641b1d42-a125-4ddb-a2b7-ec35df6f1afc" />

Your branch:

- Selected field:
<img width="1877" height="803" alt="image" src="https://github.com/user-attachments/assets/021fe978-1478-4d53-a7ec-42281eb37e70" />

- Clicked on `Debug expression`:
<img width="1877" height="803" alt="image" src="https://github.com/user-attachments/assets/65e45601-83ec-4157-9f21-2fcd7fd82a0d" />


Am I missing something here?

Commit:
fix(outputs): improve handling of array values and formatting in Wrapper.vue

Commit:
fix(outputs): enhance processedValue and transform functions for better array and object handling

iitzIrFan commented:
@MilosPaunovic sir  I attempted to create the flow using the [array-of-object.json](https://github.com/user-attachments/files/21833785/array-of-object.json)
 file from @loicmathieu sir it requires a minor adjustment, and I have updated processedValue and transform to correctly handle nested JSON, arrays, objects, and toolExecutions for accurate hierarchical display in the cascader.
 
 Updated json:
 `{
  "id": "1054zeVYYDXgupYUJZ7yu1",
  "namespace": "company.ai",
  "flowId": "ai-agent-with-tools",
  "flowRevision": 29,
  "taskRunList": [
    {
      "id": "OKNGWe7G8YyCTr723Xqst",
      "executionId": "1054zeVYYDXgupYUJZ7yu1",
      "namespace": "company.ai",
      "flowId": "ai-agent-with-tools",
      "taskId": "ai-agent",
      "attempts": [
        {
          "state": {
            "current": "SUCCESS",
            "histories": [
              {
                "state": "CREATED",
                "date": "2025-08-18T10:47:01.827363Z"
              },
              {
                "state": "RUNNING",
                "date": "2025-08-18T10:47:01.827368Z"
              },
              {
                "state": "SUCCESS",
                "date": "2025-08-18T10:47:05.997026Z"
              }
            ],
            "startDate": "2025-08-18T10:47:01.827363Z",
            "endDate": "2025-08-18T10:47:05.997026Z",
            "duration": "PT4.169663S"
          },
          "workerId": "0b6ec235-48df-42e1-ae44-a2f5295d0a14"
        }
      ],
      "outputs": {
        "completion": "The current time in UTC is 10:47:05 on 2025-08-18.",
        "tokenUsage": {
          "inputTokenCount": 605,
          "totalTokenCount": 687,
          "outputTokenCount": 45
        },
        "finishReason": "STOP",
        "toolExecutions": [
          {
            "result": {
              "timezone": "UTC",
              "datetime": "2025-08-18T10:47:05+00:00",
              "is_dst": false
            },
            "requestName": "get_current_time",
            "requestArguments": {
              "timezone": "UTC"
            }
          }
        ],
        "intermediateResponses": [
          {
            "id": "qQSjaIWFENGdvdIPhaz42AM",
            "tokenUsage": {
              "inputTokenCount": 259,
              "totalTokenCount": 313,
              "outputTokenCount": 17
            },
            "finishReason": "TOOL_EXECUTION",
            "toolExecutionRequests": [
              {
                "name": "get_current_time",
                "arguments": {
                  "timezone": "UTC"
                }
              }
            ]
          }
        ]
      },
      "state": {
        "current": "SUCCESS",
        "histories": [
          {
            "state": "CREATED",
            "date": "2025-08-18T10:47:01.613203Z"
          },
          {
            "state": "RUNNING",
            "date": "2025-08-18T10:47:01.823754Z"
          },
          {
            "state": "SUCCESS",
            "date": "2025-08-18T10:47:06.004297Z"
          }
        ],
        "startDate": "2025-08-18T10:47:01.613203Z",
        "endDate": "2025-08-18T10:47:06.004297Z",
        "duration": "PT4.391094S"
      }
    }
  ],
  "inputs": {
    "prompt": "What time is it in UTC?\r\n"
  },
  "labels": [
    {
      "key": "system.correlationId",
      "value": "1054zeVYYDXgupYUJZ7yu1"
    }
  ],
  "state": {
    "current": "SUCCESS",
    "histories": [
      {
        "state": "CREATED",
        "date": "2025-08-18T10:47:01.447739Z"
      },
      {
        "state": "RUNNING",
        "date": "2025-08-18T10:47:01.638174Z"
      },
      {
        "state": "SUCCESS",
        "date": "2025-08-18T10:47:06.118353Z"
      }
    ],
    "startDate": "2025-08-18T10:47:01.447739Z",
    "endDate": "2025-08-18T10:47:06.118353Z",
    "duration": "PT4.670614S"
  },
  "originalId": "1054zeVYYDXgupYUJZ7yu1",
  "deleted": false,
  "metadata": {
    "attemptNumber": 1,
    "originalCreatedDate": "2025-08-18T10:47:01.447773Z"
  },
  "scheduleDate": "2025-08-18T10:47:01.410Z",
  "traceParent": "00-52c248e70be9229008eb6b5cf04556f2-8dac671251ce6012-01"
}
`

iitzIrFan committed:
Merge branch 'develop' into fix/ui-array-outputs-rendering

MilosPaunovic commented:
Hi @iitzIrFan 
Can you provide screenshots/short video of how your implemented change differes in rendering from what is already on `develop` branch?

iitzIrFan committed:
Merge branch 'develop' into fix/ui-array-outputs-rendering

iitzIrFan commented:
I wanted to confirm if I am heading in the right direction, as I am a bit unsure at the moment. Could you please check if I’ve created the flow correctly? Here’s what I have done:
[test flow.json](https://github.com/user-attachments/files/22724776/test.flow.json)

MilosPaunovic committed:
Merge branch 'develop' into fix/ui-array-outputs-rendering

MilosPaunovic commented:
Let's follow the thread in the issue itself for more details https://github.com/kestra-io/kestra/issues/10770#issuecomment-3376303623.

iitzIrFan commented:
Sure @MilosPaunovic

MilosPaunovic committed:
Merge branch 'develop' into fix/ui-array-outputs-rendering

MilosPaunovic committed:
Merge branch 'develop' into fix/ui-array-outputs-rendering

MilosPaunovic committed:
Merge branch 'develop' into fix/ui-array-outputs-rendering

MilosPaunovic committed:
Merge branch 'develop' into fix/ui-array-outputs-rendering

MilosPaunovic commented:
Closed due to https://kestra-io.slack.com/archives/C03FEC452NQ/p1760703935040599?thread_ts=1760702818.651379&cid=C03FEC452NQ.
